### ISSUE TITLE
headnode/image uid/gid mismatch

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#164]
**Reported by bli on 4 Jan 2006 04:34 UTC**
```
Currently OSCAR blindly syncs (overwrite) the compute
node's/image's passwd/group files based on what's on
the headnode, this has the potential problem of making
the gid/uid of certain users out of sync.

For example if a RPM is installed on the headnode which
creates a user, but this RPM is not installed on the
compute node, having the headnode's passwd/group file
overwrite the compute node's would cause certain
file/directory ownership mismatch.

The first known "victim" of this issue is the PVM
package.  See bug #1368604 "/opt/pvm3 directory
ownership on image" for more detail.
```

**Comment by dikim33** [comment]
**Comment by bli on 31 Jul 2006 21:04 UTC**
See also #158.

**Comment by dikim33** [comment]
**Comment by efocht on 1 Aug 2006 17:57 UTC**
I take it as I'd like it fixed in 5.0.

**Comment by dikim33** [comment]
**Comment by efocht on 25 Oct 2006 17:54 UTC**
New sync_files templates handling should solve this issue (svn r5452).
Needs testing!

**Comment by dikim33** [comment]
**Comment by efocht on 25 Oct 2006 23:31 UTC**
(In [5455]) Merged commit r5452 by efocht from 2006-10-25 19:53:30 +0200 (Wed, 25 Oct 2006)
with svnmerge_1by1 version v1.1 31.7.2006
from repository https://svn.oscar.openclustergroup.org/svn/oscar/trunk
to repository https://svn.oscar.openclustergroup.org/svn/oscar/branches/branch-5-0
Original commit message was:

Updated sync_files to latest version with improved template handling.
Added automatic image template generation after image build.
Should solve #166 (master/image uid/gid mismatch).

Closes #166.
